### PR TITLE
[Fix-16331] Fix the resource file cannot be displayed hierarchically

### LINKED ISSUE #16331: [Bug] [dolphinscheduler-storage-hdfs] When you create a new task, the resource file cannot be displayed hierarchically
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

image
When creating a new task, the resource file cannot be displayed in a hierarchical manner, making it difficult to distinguish which file it is

### What you expected to happen

image
Ability to display files hierarchically

### How to reproduce

IN HdfsStorageOperator, 
```java
    public List<StorageEntity> listFileStorageEntityRecursively(String resourceAbsolutePath) {
        exceptionIfPathEmpty(resourceAbsolutePath);

        List<StorageEntity> result = new ArrayList<>();

        LinkedList<String> foldersToFetch = new LinkedList<>();
        foldersToFetch.addLast(resourceAbsolutePath);

        while (!foldersToFetch.isEmpty()) {
            String absolutePath = foldersToFetch.pollFirst();
            RemoteIterator<LocatedFileStatus> remoteIterator = fs.listFiles(new Path(absolutePath), true);
            while (remoteIterator.hasNext()) {
                LocatedFileStatus locatedFileStatus = remoteIterator.next();
                result.add(transformFileStatusToResourceMetadata(locatedFileStatus));
            }
        }
        return result;
    }
```

### Anything else

_No response_

### Version

dev

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by SbloodyS**
## Purpose of the pull request

fix #16331
fix #16336

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

## Pull Request Notice
[Pull Request Notice](https://github.com/apache/dolphinscheduler/blob/dev/docs/docs/en/contribute/join/pull-request.md)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

File: dolphinscheduler-storage-plugin/dolphinscheduler-storage-hdfs/src/test/java/org/apache/dolphinscheduler/plugin/storage/hdfs/LocalStorageOperatorTest.java [hunk 1]
```diff
@@ -296,14 +297,20 @@ public void testListStorageEntityRecursively_directory() {
...
-        assertThat(storageEntity2.getPfullName()).isEqualTo("file:" + Paths.get(baseDir, "sqlDirectory"));
-        assertThat(storageEntity2.isDirectory()).isFalse();
-        assertThat(storageEntity2.getType()).isEqualTo(ResourceType.FILE);
+        assertThat(storageEntities.size()).isEqualTo(3);
+
+        AtomicReference<StorageEntity> storageEntity2 = new AtomicReference<>();
+
+        storageEntities.stream()
+                .filter(storageEntity -> storageEntity.getFileName().equals("demo.sql"))
+                .findFirst()
```
Comment (ruanwenjun) [inline dolphinscheduler-storage-plugin/dolphinscheduler-storage-hdfs/src/test/java/org/apache/dolphinscheduler/plugin/storage/hdfs/LocalStorageOperatorTest.java]:
```suggestion
        StorageEntity storageEntity2 = storageEntities.stream()
                .filter(storageEntity -> storageEntity.getFileName().equals("demo.sql"))
                .findFirst()
                .get();
```

Comment (SbloodyS) [reply dolphinscheduler-storage-plugin/dolphinscheduler-storage-hdfs/src/test/java/org/apache/dolphinscheduler/plugin/storage/hdfs/LocalStorageOperatorTest.java]: Fixed.

**Review by ruanwenjun — APPROVED** [review]
LGTM

**Review by zhongjiajie — APPROVED** [review]
lgtm

**Pull request merged**